eclass/toolchain-funcs.eclass

```diff
@@ -421,6 +421,49 @@ tc-has-openmp() {
 	return ${ret}
 }
 
+# @FUNCTION: tc-ensure-openmp
+# @DESCRIPTION:
+# If 'openmp' is in IUSE and enabled OR if 'openmp' is not in IUSE, test
+# whether the currently active compiler supports OpenMP and die if it does
+# not. If it dies, it gives a detailed error message to the user, warning
+# them of the failure. This function should be used in pkg_setup (and
+# optionally in pkg_pretend).
+tc-ensure-openmp() {
+	case "${EAPI:-0}" in
```

Comment:
Is there a reason for the EAPI restriction? It feels kinda weird to have it inside a function.